packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts

```diff
@@ -80,26 +172,56 @@ export class DomainName extends Resource implements IDomainName {
   public readonly name: string;
   public readonly regionalDomainName: string;
   public readonly regionalHostedZoneId: string;
+  private readonly domainNameConfigurations = new Array<CfnDomainName.DomainNameConfigurationProperty>();
 
   constructor(scope: Construct, id: string, props: DomainNameProps) {
     super(scope, id);
 
+    // domain name null check
     if (props.domainName === '') {
       throw new Error('empty string for domainName not allowed');
     }
 
+    // domain name configuration null check
+    if (!props.domainNameConfigurations) {
+      throw new Error('empty domain name configurations are not allowed');
+    } else {
+      this.setDomainNameConfigurations(...props.domainNameConfigurations);
+    }
+
+    const mtlsConfig = this.configureMTLS(props.mtls);
+
     const domainNameProps: CfnDomainNameProps = {
       domainName: props.domainName,
-      domainNameConfigurations: [
-        {
-          certificateArn: props.certificate.certificateArn,
-          endpointType: 'REGIONAL',
-        },
-      ],
+      domainNameConfigurations: this.domainNameConfigurations,
+      mutualTlsAuthentication: mtlsConfig,
     };
     const resource = new CfnDomainName(this, 'Resource', domainNameProps);
     this.name = resource.ref;
     this.regionalDomainName = Token.asString(resource.getAtt('RegionalDomainName'));
     this.regionalHostedZoneId = Token.asString(resource.getAtt('RegionalHostedZoneId'));
   }
+
+  private setDomainNameConfigurations(...domainNameConfigurations: DomainNameConfiguration[]) {
+    domainNameConfigurations.forEach( (config) => {
```

Comment:
Should probably validate that endpointType is unique? That's the one constraint we do enforce for DomainNameConfigurations now, though we may add more in the future

Reply:
Do we need to do all these checks in the CDK code? I thought we could just let our service code handle all the nit-picky things.

Reply:
Shouldnt service perform that validation? We leave validation as much as possible to the service. Or in this case we should not add validations here. CFN transformation would take care of it anyway.

Reply:
Not adding any more validations. I think those need to be handled by our service code.